src/Task/Exec.php

```diff
@@ -47,11 +49,15 @@ protected function ensureValidPhappEnvironment() {
     if ($env_command = $this->manifest->getCommand('environment')) {
       $command = trim($env_command) . ' ' . $command ;
     }
-    $process = new Process($this->ensureCommandRunsViaBash($command));
+    $env = [];
+    if ($phapp_env = getenv('PHAPP_ENV')) {
+        $env['PHAPP_ENV'] = $phapp_env;
```

Comment:
should not be needed. let's remove special-casing of PHAPP_ENV here.

Reply:
```
$ phapp setup localdev
 [CollectionBuilder]    in task Robo\Collection\CompletionWrapper 

  The phapp environment is undefined. Run "phapp setup $ENVIRONMENT" to initialize it. 
 [CollectionBuilder]  Exit code 1 
 [notice] Stopping on fail. Exiting....
 [error]  Exit Code: 1 
 [error]    in task Robo\Collection\CollectionBuilder 

    in task Robo\Collection\CompletionWrapper 

  The phapp environment is undefined. Run "phapp setup $ENVIRONMENT" to initialize it. 
```

Reply:
for setup command PHAPP_ENV (in bash) is required so having a check for that should not hurt us. There is this custom override of environment command in setup command (which just clears environment command manifest) but the check is still run.

To further support my findings there is also this method:
```
  protected function initShellEnvironment() {
    // Switch working directory.
    chdir($this->phappManifest->getFile()->getPath());
    // Add the composer bin-dir to the path.
    $path = getenv("PATH");
    putenv("PATH=../vendor/bin/:../bin:$path");
    return $this;
  }
```
This looks like there is another php env var that is supposed to be transfered to the bash variables.

Reply:
agree